Library/CSharpWrapper/src/NativeClassWrapper.cs

```diff
@@ -38,5 +38,52 @@ internal NativeClassWrapper(NativePointer ptr)
             _ptr = ptr.Pointer;
             _ownsPtr = ptr.OwnsOwnData;
         }
+
+        protected virtual void DisposeNativeObject() { }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
+            {
+                try
+                {
+                    if (_ownsPtr)
+                    {
+                        if (!disposing)
+                        {
+                            //TODO: Log that the object wasn't disposed and is being destroyed by finalizer
+                        }
+
+                        if (_ptr == IntPtr.Zero)
+                        {
+                            throw new InvalidOperationException("Internal pointer was null");
+                        }
+
+                        if (CSPFoundation.GetIsInitialised())
```

Comment:
This line makes Unity tests fail since `CSPFoundation` is a generated class that doesn't exist in those tests. There might be a more elegant way to handle this.